corr-app/correlationDataHandler.py

```diff
@@ -0,0 +1,133 @@
+            df['vwap'] = (df["Volume"] * df["Close"]) / df["Volume"]
+            df['returns'] = df["Close"] / df.Close.shift(1) - 1
+            df.insert(0, "Ticker", ticker)
+            results = pd.concat([results, df])
+            
+        return results
+    
+    def clean_frame(self, period) -> pd.DataFrame:
+        """
+        Clean up the data 
+        """
+        
+        df = self.build_frame(period)
+        
+        df = df.loc[:, ["Ticker", "vwap"]]
+        
+        df.dropna(axis = 0, how = "all", inplace = True)
+        
+        df["date"] = pd.to_datetime(df.index, utc = True).date
+        
+        df.reset_index(drop = False, inplace = True)
+        
+        df = pd.pivot(df, index = "date", columns = "Ticker", values = "vwap")
+        
+        return df
+    
+    def correlation_links(self, period) -> pd.DataFrame:
+        """
+        Get the correlations for a given time period
+        """
+    
+        df = self.clean_frame(period)
+        
+        df = df.rename_axis(None).rename_axis(None, axis=1).corr()
+        df = df.stack().reset_index()
+        df = pd.DataFrame(df)
+        df.columns = ['source', 'target', 'Correlation']
+        df = df[df['Correlation'] != 1]
+        
+        df = df.drop_duplicates()
+
+        df["oneway"] = df.apply(lambda x: not df[
+            (df["source"] == x["target"]) & (df["target"] == x["source"]) & (df.index != x.name)].empty,axis=1)
+
+        df = df[df['oneway'] == True].drop("oneway", axis = 1)
+
+        # normalize correlation
+        df["norm_corr"] = ((df["Correlation"] - df["Correlation"].mean()) / df["Correlation"].std()).round(3)
+        
+        #df = df.groupby(['source'], as_index = False)['Correlation'].mean().round(3).rename(columns={'source':'name'})
+
+        #df = df.drop_duplicates()
+        
+        df["period"] = period
+        
+        return df
+    
+    def correlation_nodes(self, df) -> pd.DataFrame:
+        """
+        Get the average correlations 
+        """
+        
+        df = df.groupby(['source'], as_index = False)['Correlation'].mean().round(3).rename(columns={'source':'name'}).drop_duplicates()
+        
+        return df
+    
+#     def get_all_correlations(self) -> pd.DataFrame:
+#         """
+#         loop through all time period of interest and get the correlations into one usable dataframe
+#         """
+        
+#         periods = ["1mo", "3mo", "6mo", "1y", "2y"]
+#         results = pd.DataFrame()
+        
+#         for period in periods:
+#             df = self.period_correlations(period)
+            
+#             results = pd.concat([results, df])
+            
+#         results.reset_index(drop = True)
+        
+#         return results
+    
+    def get_securities(self) -> list:
+        """
+        shows the current securities list
+        """
+        
+        return self.ticker_list
+    
+    def remove_security(self, security_to_remove) -> list:
```

Comment:
I like these add and remove security functions. They're simple, and well named! Nice job thinking to specify the object returned. In this case, the function returns a list in one case, and a string in the other. I don't know if we can specify a type hint for alternative cases like that. My initial though is that this function should probably return a list every time, and maybe we can adjust the if/else logic in a way so we don't ever have to return a string.

I am not trying to over-complicate this. My intuition is that we shouldn't have a function that returns a list or returns a print. Maybe we can do 5 minutes of research and understand if this approach is okay or could be better.

One suggestion, for 'security_to_remove', you could add a type hint to specify that the value should be a string.